src/pybind/mgr/dashboard_v2/services/ceph_service.py

```diff
@@ -1,32 +1,112 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
-from . import Service
+import json
+import rbd
 
+from ..tools import ViewCache
+from .. import logger
+
+
+def _get_service_map(mgr, service_name):
+    service_map = {}
+    for server in mgr.list_servers():
+        for service in server['services']:
+            if service['type'] == service_name:
+                if server['hostname'] not in service_map:
+                    service_map[server['hostname']] = {
+                        'server': server,
+                        'services': []
+                    }
+                inst_id = service['id']
+                metadata = mgr.get_metadata(service_name, inst_id)
+                status = mgr.get_daemon_status(service_name, inst_id)
+                service_map[server['hostname']]['services'].append({
+                    'id': inst_id,
+                    'type': service_name,
+                    'hostname': server['hostname'],
+                    'metadata': metadata,
+                    'status': status
+                })
+    return service_map
+
+
+def _get_pool_list(mgr, application=None):
+    osd_map = mgr.get('osd_map')
+    if not application:
+        return osd_map['pools']
+    return [pool for pool in osd_map['pools']
+            if application in pool.get('application_metadata', {})]
+
+
+def _get_daemons(mgr):
```

Comment:
The name is misleading in the context of `ceph_service.py`: It doesn't return generic daemons, instead it is tightly bound to the rbd-mirroring controller